.github/workflows/deploy.yaml

```
@@ -125,6 +126,63 @@ jobs:
       - id: arm64
         if: ${{ inputs.arm-build }}
         run: echo "runner=self-hosted-arm64" >> $GITHUB_OUTPUT
+      - id: verify-argocd-token
+        if: ${{ ! inputs.push-to-manifests }}
+        shell: bash
+        run: |
+          if [ "${{ secrets.ARGOCD_TOKEN }}" == '' ]; then
+            echo "MUST INPUT ARGOCD_TOKEN SECRET"
+            echo "ARGOCD_TOKEN_STAGING / ARGOCD_TOKEN_PRODUCTION"
+            exit 1
+          fi
+      - id: set-argocd-server
+        if: ${{ env.argocd-token != '' }}
+        env:
+          STAGE: ${{ inputs.stage }}
+        run: |
+          if [[ $STAGE =~ dev|staging ]]; then
+            ARGOCD_HOSTNAME="argocd.staging.monta.app"
+          else
+            ARGOCD_HOSTNAME="argocd.monta.app"
+          fi
+          echo "argocd-hostname=$ARGOCD_HOSTNAME" >> "$GITHUB_OUTPUT"
+      - name: Get manifest info
+        id: get-manifest-info
+        if: ${{ inputs.upload-open-api }}
+        shell: bash
+        env:
+          SERVICE_IDENTIFIER: ${{ inputs.service-identifier }}
+          STAGE: ${{ inputs.stage }}
+          ARGOCD_TOKEN: ${{ secrets.ARGOCD_TOKEN }}
+          ARGOCD_SERVER: ${{ steps.set-argocd-server.outputs.argocd-hostname }}
+        run: |
+          if [ "${{ secrets.ARGOCD_TOKEN }}" == '' ]; then
+            echo "MUST INPUT ARGOCD_TOKEN SECRET"
```

```suggestion
            echo "::error::MUST INPUT ARGOCD_TOKEN SECRET"
```

All look great. If I have to say something, [workflow commands](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions) will be it. It is just a visual improvement over stdout, giving you a red error message.

---

Oh that's amazing, didn't know about that!